File: 1.21.6/src/main/java/com/bleudev/nine_lifes/networking/payloads/BetaModeMessage.java
package com.bleudev.nine_lifes.networking.payloads;

import com.bleudev.nine_lifes.networking.Packets;
import net.minecraft.network.RegistryByteBuf;
import net.minecraft.network.codec.PacketCodec;
import net.minecraft.network.packet.CustomPayload;

public record BetaModeMessage() implements CustomPayload {
    public static final Id<BetaModeMessage> ID = new Id<>(Packets.BETA_MODE_MESSAGE);
    public static final PacketCodec<RegistryByteBuf, BetaModeMessage> CODEC = PacketCodec.unit(new BetaModeMessage());

    @Override
    public Id<? extends CustomPayload> getId() {
        return ID;
    }
}

File: 26.3/src/main/kotlin/com/bleudev/nine_lifes/custom/packet/payload/unit/BetaModeMessage.kt
package com.bleudev.nine_lifes.custom.packet.payload.unit

import com.bleudev.nine_lifes.custom.NineLifesPackets
import com.bleudev.nine_lifes.custom.packet.payload.interfaces.PacketPayloadCompanion
import net.minecraft.network.RegistryFriendlyByteBuf
import net.minecraft.network.codec.StreamCodec
import net.minecraft.network.protocol.common.custom.CustomPacketPayload

class BetaModeMessage : CustomPacketPayload {
    private constructor()

    companion object : PacketPayloadCompanion<BetaModeMessage> {
        val INSTANCE = BetaModeMessage()

        override val idLocation = NineLifesPackets.UNIT_BETA_MODE_MESSAGE
        override val codec: StreamCodec<RegistryFriendlyByteBuf, BetaModeMessage> = StreamCodec.unit(INSTANCE)
    }

    override fun type(): CustomPacketPayload.Type<out CustomPacketPayload> = id
}